tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/upgrade/UpgradeAssistantSettingsUtil.java

```diff
@@ -85,6 +85,22 @@ public boolean accept( File dir, String name )
         return null;
     }
 
+    public static <T> T getObjectFromStore( Class<T> clazz, String suffix ) throws IOException
```

Comment:
for 2,if we don't use "suffixes" it will be all of date will be saved in one json file, and if we do every "update" actions for only single problem it will "update" whole of json file, I think it will take some Unnecessary time on it.